### ISSUE TITLE
S2583 FN: Local functions

### ISSUE THREAD

**Issue by hankovich** [SonarSource/sonar-dotnet#4515]
### Description

```csharp
public void A()
{
    int? i = GetNullableInt();

    if (i == null && i == 3) // <- S2583 is reported here (i == 3 is always false)
    {

    }

    static void B()
    {
        int? i = GetNullableInt();

        if (i == null && i == 3) // <- but... S2583 is NOT reported here
        {

        }
    }

    static int? GetNullableInt() => 42;
}
```

### Related information

* C#/VB.NET Plugins version: 8.23.0.32424

**Comment by csaba-sagi-sonarsource** [comment]
Hi @hankovich , I can confirm the false negative. Thanks for your feedback.

**Comment by mary-georgiou-sonarsource** [comment]
Fixed by https://github.com/SonarSource/sonar-dotnet/pull/7478
